### ISSUE TITLE
How to migrate from custom component to native support

### ISSUE THREAD

**Issue by JanZimmer** [tschamm/boschshc-hass#28]
Congrats on being accepted as an official hass integration :)

Is there anything that has to be done before updating home assistant to the most current version for those who are already using the custom integration?

Thanks!

**Comment by tschamm** [comment]
Thanks! 😊
You can have both integrations, the official as well as the custom component, installed in parallel. Thus you can just update to 2021.6.
As there are still many features missing in the official integration, I suggest you‘ll keep the custom component installed.

**Comment by therealhalifax** [comment]
> 
> 
> Thanks! 😊
> You can have both integrations, the official as well as the custom component, installed in parallel. Thus you can just update to 2021.6.
> As there are still many features missing in the official integration, I suggest you‘ll keep the custom component installed.
> 
> I'll keep this issue open and will put a guide here, as soon as features are mostly merged into the official integration.

With the update to 2021.7 came up more sensors, but I am not sure if this is everything? Will also the Heater control work?

**Comment by therealhalifax** [comment]
How to use both in parallel? I try today, but there came up "Device already configured". And i see only one integration called Bosch SHC. I assume thats the custom one?

**Comment by tschamm** [comment]
When you have the custom component installed, you cannot use the official in parallel. Have a look into the full log to see if the custom component is used. At the beginning of the logs it should show something like: 
`We found a custom integration bosch_shc which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant`.

**Comment by therealhalifax** [comment]
@tschamm But above you wrote that both in parallel. Yes i am sure to use the custom one. There is also a small icon shown which indicates that it is a custom integration. 

anyway, unitl now I am fine with the custom one, I just wanted to give the core one a try since I am faced with a bug with the last HA update. 

Hope you can fix that, I really like your integration. It was also the main reason for me to switch to Home Assistant.

**Comment by tschamm** [comment]
@therealhalifax That was misleading, I meant you can have installed both in parallel. However, if the custom component is installed, it will be prioritized over the official integration. Thus, if you want to use the offical integration, you'll have to remove the custom component.

**Comment by Matze89x** [comment]
Hello to you and hello to @tschamm, 

I think it's great because I have completely stocked up on Bosch Smart Home products in the entire household and I use this plugin with good certainty.

Now I'm wondering when this plugin will be available in the official version or are there problems with the release of updates? 

I am only noticing this more and more, as I have already caught up with many colleagues who are also using Bosch Smart Home with Home Assistant and the topic keeps coming up as to why this integration is not already officially delivered with Home Assistant. 

That's why I wanted to find out here.
Have a nice day and thank you very much.

**Comment by mosandlt** [comment]
The HACS custom component and the built-in HA Core integration share the same `bosch_shc` domain — when the HACS version is installed it takes precedence, so there's nothing special to migrate. The HACS build is simply the more up-to-date one and is what most users run.

Closing as answered — happy to reopen if you hit a concrete migration problem.
